tgui/packages/tgui/interfaces/ExodroneConsole.tsx

```diff
@@ -0,0 +1,793 @@
+import { formatTime } from '../format';
+import { capitalize } from 'common/string';
+import { map, toKeyedArray } from '../../common/collections';
+import { IconStack } from '../components/Icon';
+import nt_logo from '../assets/bg-nanotrasen.svg';
+import { MiningVendor } from './MiningVendor';
+
+
+type ExplorationEventData = {
+  name: string,
+  ref: string
+}
+
+type FullEventData = {
+  image: string,
+  description: string,
+  action_enabled: boolean,
+  action_text: string,
+  skippable: boolean,
+  ignore_text: string,
+  ref: string
+}
+
+type ChoiceData = {
+  key: string,
+  text: string
+}
+
+type AdventureData = {
+  description: string,
+  image: string,
+  raw_image: string,
+  choices: Array<ChoiceData>
+}
+
+type SiteData = {
+  name: string,
+  ref: string,
+  description: string,
+  coordinates: string,
+  distance: number,
+  band_info: Record<string, number>,
+  revealed: boolean,
+  point_scan_complete: boolean,
+  deep_scan_complete: boolean,
+  events: Array<ExplorationEventData>
+}
+
+
+enum DroneStatusEnum {
+  Idle = "idle",
+  Travel = "travel",
+  Exploration = "exploration",
+  Adventure = "adventure",
+  Busy = "busy"
+}
+
+enum CargoType {
+  Tool = "tool",
+  Cargo = "cargo",
+  Empty = "empty"
+}
+
+type CargoData = {
+  type: CargoType,
+  name: string
+}
+
+type DroneBasicData = {
+  name: string,
+  description: string,
+  controlled: boolean,
+  ref: string,
+}
+
+type ExodroneConsoleData = {
+  signal_lost: boolean,
+  drone: boolean,
+  all_drones?: Array<DroneBasicData>
+  drone_status?: DroneStatusEnum,
+  drone_name?: string,
+  drone_integrity?: number,
+  drone_max_integrity?: number,
+  drone_travel_coefficent?: number,
+  drone_log?: Array<string>,
+  configurable?: boolean,
+  cargo?: Array<CargoData>,
+  can_travel?: boolean,
+  travel_error: string,
+  sites?: Array<SiteData>,
+  site?: SiteData,
+  travel_time?: number,
+  travel_time_left?: number,
+  wait_time_left?: number,
+  wait_message?: string,
+  event?: FullEventData,
+  adventure_data?: AdventureData,
+  // ui_static_data
+  all_tools: Record<string, ToolData>,
+  all_bands: Record<string, string>
+}
+
+type ToolData = {
+  description: string,
+  icon: string
+}
+
+export const ExodroneConsole = (props, context) => {
+  const { data } = useBackend<ExodroneConsoleData>(context);
+  const {
+    signal_lost,
+  } = data;
+
+  const [
+    choosingTools,
+    setChoosingTools,
+  ] = useLocalState(context, 'choosingTools', false);
+
+  return (
+    <Window width={650} height={500}>
+      {!!signal_lost && <SignalLostModal />}
+      {!!choosingTools && <ToolSelectionModal />}
+      <Window.Content>
+        <ExodroneConsoleContent />
+      </Window.Content>
+    </Window>
+  );
+};
+
+const SignalLostModal = (props, context) => {
+  const { act } = useBackend(context);
+  return (
+    <Modal backgroundColor="red" textAlign="center" width={30} height={22} p={0} style={{ "border-radius": "5%" }}>
```

Comment:
each prop should be on its own line (so bring it to a new line and indent it once)